### PR TITLE
Fixes#497 Added gh-pages deployment for frontend

### LINKED ISSUE #497: Deploy backend and front-end to heroku.
**I'm submitting a ...**
- [ ] bug report
- [x] feature request

**Current behavior:**

As the backend and front end were seperate we dont have heroku deployment for both

**Expected behavior:**

Front end and backend should be deploy ee to heroku. They must be two heroku apps.

So we need to:
- [ ] Deploy front-end to gh-pages
- [ ] Deploy backend to heroku
- [ ] Update the form action in the front-end to the heroku deployment url of the backend.

At the end we should get a running app that is alle to generate badges and provide all functionality that v1 supplies. (Code is already there)

Note: Proceed only after an admin like @mariobehling gives you a url for the front-end and backend to be on heroku. After deployment it will be better if the ownership is transferred to him as well.

### PR DESCRIPTION
**Pull request by ParthS007**
Fixes #497 

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] I have added necessary documentation (if appropriate)

#### Changes proposed in this pull request:

- Added gh-pages deployment for frontend

### CODE REVIEW

**Comment by ParthS007** [comment]
@harshit98 @abishekvashok @vaibhavsingh97 @S2606 @rupav  Please review this and suggest changes, I will update PR after your reviews. Thanks

**Comment by ParthS007** [comment]
@harshit98 @abishekvashok @S2606 @vaibhavsingh97 @rupav  Please review

File: frontend/config/environment.js [hunk 1]
```diff
@@ -43,8 +38,13 @@ module.exports = function(environment) {
     ENV.APP.autoboot = false;
   }
 
+  var deployTarget = process.env.DEPLOY_TARGET;
```
[the remark this answers is not in the document]
Comment (gabru-md) [reply frontend/config/environment.js]: @ParthS007 you have defined `process` as a global in `.eslintrc.js` bu Hound cannot find it.
Please look into it

**Comment by abishekvashok** [comment]
@ParyhS007 can you check why process is not defined as stated by hound

**Comment by ParthS007** [comment]
@abishekvashok  I am not able to figure out this, I think this can be ignored

File: .travis.yml [hunk 2]
```diff
@@ -61,7 +61,10 @@ matrix:
       - npm i eslint-plugin-ember@latest --save-dev
       - npm config set spin false
 
-
     script:
       - npm run lint:js
       - npm test
+
+    after_success:
+      - chmod 777 ./scripts/gh_deploy.sh
```
Comment (abishekvashok) [inline .travis.yml]: Run `git chmod 777 ./scripts/gh_deploy.sh` that will change the access parameters for the file. Then add it and commit. git will retain file permissions. And then we can remove this safely.

Comment (ParthS007) [reply .travis.yml]: @abishekvashok  `chmod ` is not a git command. You can update it by `git update-index --chmod=+x ./scripts/gh_deploy.sh` .

**Comment by ParthS007** [comment]
@abishekvashok @gabru-md Please review now, I will configure Hound to review frontend in #533.

File: frontend/script/gh_deploy.sh [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+git config --global user.email "noreply+travis@fossasia.org"
+
+export DEPLOY_BRANCH=${DEPLOY_BRANCH:-development}
+
+if [ "$TRAVIS_PULL_REQUEST" != "false" -o "$TRAVIS_REPO_SLUG" != "fossasia/badgeyay" -o "$TRAVIS_BRANCH" != "$DEPLOY_BRANCH" ]; then
+    echo "Skipping deploy; The request or commit is not on development:)"
+    exit 0
+fi
+
+ember deploy gh-pages
```
Comment (abishekvashok) [inline frontend/script/gh_deploy.sh]: Sorry for not mentioning earlier. Okay, won't we need a commit?

Comment (ParthS007) [reply frontend/script/gh_deploy.sh]: @abishekvashok  I have made depoy.js which will handle all these things.

Comment (gabru-md) [reply frontend/script/gh_deploy.sh]: I guess it is checking for the branch, is the branch is not `dev` then there is no need to deploy again?
Ping @abishekvashok

**Pull request merged**